modules/sitemaps/sitemaps.php

```diff
@@ -451,10 +458,12 @@ public function callback_action_flush_news_sitemap_cache() {
 	 *
 	 * @access public
 	 * @since 5.3.0
+	 * @since 6.7.0 Schedules a regeneration after 60 seconds.
 	 */
 	public function callback_action_purge_data() {
 		$this->callback_action_flush_news_sitemap_cache();
 		$this->librarian->delete_all_stored_sitemap_data();
+		wp_schedule_single_event( time() + 60, 'jp_sitemap_cron_hook' );
```

Comment:
Should we randomly spread this across 5mins or so to avoid heavy server load in situations where Jetpack is upgraded for multiple sites on same server?

Reply:
You're absolutely right. I'll update that in my morning.

Reply:
Can we make the random distribution configurable? I'm sure that some hosts that upgrade Jetpack centrally would appreciate being able to spread the load over, say, an hour.

Reply:
How about like this? https://github.com/Automattic/jetpack/pull/10274/commits/44f86066e2d7787cce364480fd042fecfa2cea0b

(I'll introduce the same thing here too)